### PR TITLE
Handle pictures from other repos

### PR DESCRIPTION
**Pull request by annatraussnig**

### CODE REVIEW

File: packages/publikator/graphql/resolvers/_mutations/commit.js [hunk 1]
```diff
@@ -28,33 +29,61 @@ const {
...
     }
   }
   return url
 }
 
+const isFromRepo = (url, repoId) =>
+  url && url.startsWith(`${ASSETS_SERVER_BASE_URL}/github/`) && !url.includes(repoId)
+
+const importFromRepo = async (url, images, repoId) => {
+  if (isFromRepo(url, repoId)) {
```
Comment (tpreusse) [inline packages/publikator/graphql/resolvers/_mutations/commit.js:74]:
```suggestion
const isFromDifferentRepo = (url, repoId) =>
  url && url.startsWith(`${ASSETS_SERVER_BASE_URL}/github/`) && !url.includes(repoId)

const importFromRepo = async (url, images, repoId) => {
  if (isFromDifferentRepo(url, repoId)) {
```

**Review by tpreusse — APPROVED** [review]
LGTM. One naming suggestion. What did we miss earlier?

File: packages/publikator/graphql/resolvers/_mutations/commit.js [hunk 2]
```diff
@@ -28,33 +29,61 @@ const {
...
+    try {
+      const blob = await fetch(url).then(result => result.buffer())
+      const { image, newUrl } = await generateImageData(blob)
+      images.push(image)
+      return newUrl
+    } catch (e) {
+      console.error('failed to transfer image', e)
```
Comment (tpreusse) [inline packages/publikator/graphql/resolvers/_mutations/commit.js:81]: will help us if we want to case it down:
```suggestion
      console.error('failed to transfer image', repoId, url, e)
```

File: packages/publikator/graphql/resolvers/_mutations/commit.js [hunk 3]
```diff
@@ -28,33 +29,61 @@ const {
...
+const {
+  ASSETS_SERVER_BASE_URL
+} = process.env
+
+const generateImageData = async (blob) => {
+  const meta = await sharp(blob).metadata()
+  // image/jpeg -> jpeg
+  // image/png -> png
+  // image/svg+xml -> svg
+  const suffix = blob.type.split('/')[1].split('+')[0]
```
Comment (tpreusse) [inline packages/publikator/graphql/resolvers/_mutations/commit.js]: got an error on staging
```
failed to transfer image republik-dev/article-test-123-copy-from https://cdn.republik.pink/github/republik-dev/article-das-kartell/images/e9372d1d0ac203f93a59ccc5d62279d1a2a61ae2.jpeg?size=3800x2533#originalURL=images%2Fe9372d1d0ac203f93a59ccc5d62279d1a2a61ae2.jpeg%3Fsize%3D3800x2533 TypeError: Cannot read property 'split' of undefined
```

`blob.type` is specific to `dataUriToBuffer` to I changed it to `meta.format`

File: packages/documents/graphql/schema-types.js [hunk 4]
```diff
@@ -74,7 +74,7 @@ type Meta {
   estimatedConsumptionMinutes: Int
 
   # template of the article
-  template: String
+  template: String!
```
Comment (annatraussnig) [inline packages/documents/graphql/schema-types.js:77]: @tpreusse I can do this or will it make problems?

**Review by tpreusse — APPROVED** [review]

**Pull request merged**